magento/magento2
# System xml cleanup

### Description (*)

This PR aims to (at least partially) cleanup system.xml files:
- `showInStore="0"`, `showInWebsite="0"`, `showInDefault="0"` entries were removed as this is default value
- some default values were added to config.xml (i.e. no value set for field with `Magento\Config\Model\Config\Source\Yesno` model was set to 0 as default (to not change current behavior)
- more fields have `canRestore="1"` set now, to make it possible to reset to default value
- more configuration sections for one specific functionality have `depends` configuration added to only show if functionality itself is enabled.

This also includes changes from #26354 - the other PR can be merged at first or just closed if necessary.


### Manual testing scenarios (*)

1. Go to `Stores -> Configuration`
2. More fields (too many changes to include every single place changed here) should have checkbox to restore default value
3. More functionality configurations should be visible only if functionality is enabled (i.e. Newsletter)

### Questions or comments


### Contribution checklist (*)
